### PR TITLE
Add xgboost

### PR DESCRIPTION
**Pull request by CAM-Gerlach**
Per the discussion in dmlc/xgboost#4379

Just what it says on the tin, per the consent of the maintainers, as they're dropping Python 2 support in their imminent next release. @hcho3 please review and sign off.

### CODE REVIEW

File: site.js [hunk 1]
```diff
@@ -128,6 +128,10 @@ $(document).ready(function (){
       {content: 'Python 2 & 3', start: '2010-03-25', end:'2019-03-18', py2:true},
       {content: 'Python 3 only', start: '2019-03-19', end:'2023-12-31'},
     ],
+    'XGBoost':[
+      {content: 'Python 2 & 3', start: '2016-01-14', end:'2019-05-15', py2:true},
+      {content: 'Python 3 only', start: '2020-05-15', end:'2023-12-31'},
```
Comment (hcho3) [inline site.js]:
```suggestion
      {content: 'Python 3 only', start: '2019-05-15', end:'2023-12-31'},
```
Is this a typo?

Comment (CAM-Gerlach) [reply site.js]: Yep, thanks X( Fixed.

Comment (hcho3) [reply site.js]: Thanks!

**Comment by CAM-Gerlach** [comment]
@takluyver Looks like this is ready.

**Comment by takluyver** [comment]
Thanks!

**Pull request merged**